opte/src/engine/icmpv6.rs

```diff
@@ -172,3 +176,296 @@ impl HairpinAction for Icmpv6EchoReply {
+        let (src_ip, dst_ip) = if let Some(metadata) = meta.inner_ip6() {
+            (
+                IpAddress::Ipv6(Ipv6Address(metadata.src.bytes())),
+                IpAddress::Ipv6(Ipv6Address(metadata.dst.bytes())),
+            )
+        } else {
+            // Getting here implies the predicate matched, but that the
+            // extracted metadata indicates this isn't an IPv6 packet. That
+            // should be impossible, but we avoid panicking given the kernel
+            // context.
+            return Err(GenErr::Unexpected(format!(
+                "Expected IPv6 packet metadata, but found: {:?}",
+                meta
+            )));
+        };
+        let body = rdr.copy_remaining();
+        let src_pkt = Icmpv6Packet::new_checked(&body)?;
+        let src_ndisc =
+            Icmpv6Repr::parse(&src_ip, &dst_ip, &src_pkt, &Csum::ignored())?;
+
+        if !matches!(
+            src_ndisc,
+            Icmpv6Repr::Ndisc(NdiscRepr::RouterSolicit { .. })
+        ) {
+            // We should never hit this case because the predicate
+            // should have verified that we are dealing with an
+            // Router Solicitation. However, programming error could
+            // cause this to happen -- let's not take any chances.
+            return Err(GenErr::Unexpected(format!(
+                "expected a NDP Router Solicitation, got {} {}",
+                src_pkt.msg_type(),
+                src_pkt.msg_code()
+            )));
+        }
+
+        // RFC 4861 6.1.1 describes a number of validation steps routers are
+        // required to perform.
+        if src_pkt.msg_code() != 0 {
+            return Ok(AllowOrDeny::Deny);
+        }
+        if !src_pkt.verify_checksum(&src_ip, &dst_ip) {
+            return Ok(AllowOrDeny::Deny);
+        }
+        // NOTE: The router is required to check that there is no Link-Layer
+        // Address Option, if the solicitation is sent from the unspecified
+        // address. The predicates here are that the solicitation came from the
+        // guest's EUI-64 transform link-local, so that is checked.
+        //
+        // TODO-completeness: Check IP Hop Limit and ICMP length / option length
+
+        let flags = if self.managed_cfg {
+            NdiscRouterFlags::MANAGED
+        } else {
+            NdiscRouterFlags::empty()
+        };
+        let advert = NdiscRepr::RouterAdvert {
+            hop_limit: u8::MAX,
+            flags,
+            // Use the maximum advertised lifetime as a default router.
+            router_lifetime: Duration::from_secs(9_000),
+            // Do not specify the reachable or retrans time. Guests will decide
+            // that for themselves at this point.
+            reachable_time: Duration::from_millis(0),
+            retrans_time: Duration::from_millis(0),
+            lladdr: Some(RawHardwareAddress::from_bytes(&self.mac.bytes())),
+            mtu: Some(1500),
```

Comment:
Ok, thanks. I'll refer to that issue, and I agree adding the MTU into the `VpcCfg` seems like a pretty easy and valuable solution.

To make sure I understand the point about headroom though. We can advertise 1500 to clients, and I recall agreeing at some point that we were sticking to that in the MVP. In that case, ensuring headroom means using a larger MTU in `xde`, right? I'm just checking that we want to advertise 1500 to _clients_, rather than use 1500 ourselves and advertise a smaller MTU to clients.

Reply:
Yes, 1500 to clients/instances at this time. In our VPC case we decided we'd stick with 1500 for now but have the machinery to allow configuration in the future when we deem it's needed/wanted.

https://rfd.shared.oxide.computer/rfd/0021#design-mtu

So with decision in place we just need to make sure our transit links (T6 NICs and the switches) are configured with a large enough MTU to fit the VPC encap headers. This shouldn't affect traffic reaching out to the external network as the VPC encap only exists inside the rack/fleet -- so the external network shouldn't have to make any adjustments to handle the frames coming out of our switch's upstream ports.

Reply:
Yeah, that all makes sense, thanks.